kotlin-inject-compiler-core/src/main/kotlin/me/tatarka/inject/compiler/CycleDetector.kt

```diff
@@ -2,25 +2,57 @@ package me.tatarka.inject.compiler
 
 class CycleDetector {
 
-    private val _elements = mutableListOf<AstElement>()
+    private val _elements = mutableListOf<Entry>()
+    private var resolving: CycleResult.Resolvable? = null
```

Comment:
Can there be more than one lazy element resolved at a time?

Reply:
Good question, will try to figure out if I can write a test where that is the case.

Reply:
Yep, added a failing test-case and fixed it.